include/vcpkg/vcpkgpaths.h

```diff
@@ -152,7 +152,8 @@ namespace vcpkg
                                                                            const Path& relative_path_to_file) const;
         ExpectedS<Path> git_checkout_object_from_remote_registry(StringView tree) const;
 
-        Optional<const ManifestAndPath&> get_manifest() const;
+        const Optional<ManifestAndPath>& get_manifest() const;
+        const Optional<Path>& get_manifest_directory() const;
```

Comment:
After discussion with @valeriaconde I realize that adding this function is almost certainly wrong because artifacts interacts with `vcpkg-configuration.json`, not `vcpkg.json`, so we are giving it the wrong path. I should fix this to pass the path to the file to exit explicitly...